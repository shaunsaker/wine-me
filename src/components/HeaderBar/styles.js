import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    paddingTop: styleConstants.helpers.isIphoneX ? 24 : styleConstants.dimensions.spacing.vertical,
    paddingBottom: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    backgroundColor: styleConstants.colors.primary,
  },
});

export default styles;
