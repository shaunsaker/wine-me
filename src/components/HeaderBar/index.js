import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {};

const HeaderBar = ({ children }) => {
  return <View style={styles.container}>{children}</View>;
};

HeaderBar.propTypes = propTypes;
HeaderBar.defaultProps = defaultProps;

export default HeaderBar;
