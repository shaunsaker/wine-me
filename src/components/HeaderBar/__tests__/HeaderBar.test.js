import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import HeaderBar from '..';

describe('HeaderBar', () => {
  const children = <View />;
  const style = { backgroundColor: 'red' };

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<HeaderBar style={style}>{children}</HeaderBar>);

      expect(component).toMatchSnapshot();
    });
  });
});
