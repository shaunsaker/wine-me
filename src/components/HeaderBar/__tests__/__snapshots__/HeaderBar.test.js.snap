// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar renders renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#48466D",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View />
</View>
`;
